play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/StandaloneAhcWSRequest.scala

```diff
@@ -151,10 +154,9 @@ case class StandaloneAhcWSRequest(
   }
 
   private def withBodyAndContentType(wsBody: WSBody, contentType: String): Self = {
-    if (headers.contains(HttpHeaders.Names.CONTENT_TYPE)) {
-      copy(body = wsBody)
-    } else {
-      copy(body = wsBody).addHttpHeaders(HttpHeaders.Names.CONTENT_TYPE -> contentType)
+    this.contentType match {
+      case Some(_) => copy(body = wsBody)
+      case None => copy(body = wsBody).withContentType(contentType)
```

Comment:
I realize this is a refactor and the original logic hasn't changed, but -- if you set a body which has a content type, and there's already a content type on the request, why is the body's content type less valid and not appended?  Shouldn't the behavior be last writer wins?